### PR TITLE
Setup CI on 0.4.x branch

### PR DESCRIPTION
**Pull request by bernardnormier**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -55,6 +55,7 @@ jobs:
       uses: actions/checkout@master
       with:
         repository: icerpc/icerpc-csharp
+        ref: 0.4.x
```
Comment (pepone) [inline .github/workflows/ci.yml:58]: That is fine, but shouldn't we just use the published packages for branches other than main?

**Review by pepone — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (externl) [reply .github/workflows/ci.yml:58]: Good idea.

**Comment by externl** [comment]
CI is failing

File: .github/workflows/ci.yml (hunk 1 above)
Comment (bernardnormier) [reply .github/workflows/ci.yml:58]: On the contrary, I think we do want to test against the branch to make sure we don't break interop in a patch release.

Comment (externl) [reply .github/workflows/ci.yml:58]: Also a good point.

**Review by externl — APPROVED** [review]

**Pull request merged**